### PR TITLE
Add interface for archive type that does not include compressed storage

### PR DESCRIPTION
**Pull request by fireworkmarks**

### CODE REVIEW

File: ceph/cluster.go [hunk 1]
```diff
@@ -210,7 +211,7 @@ func (rd *RadosSmallDownloader) Close() error {
 	return nil
 }
 
-func (cluster *CephCluster) Put(poolname string, data io.Reader) (oid string,
+func (cluster *CephCluster) Put(poolname string, isCompressible bool, data io.Reader) (oid string,
```
Comment (cxt90730) [inline ceph/cluster.go]: ceph not need compress operation. Do it on previous level

File: integrate/yigconf/yig.toml [hunk 2]
```diff
@@ -55,6 +56,11 @@ upload_max_chunk_size = 8388608 #8MB
 ceph_config_pattern = "/etc/ceph/*.conf"
 
 # Plugin Config
+[plugins.snappy]
```
Comment (cxt90730) [inline integrate/yigconf/yig.toml]: add a dummy plugin

File: compression/const.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package compression
```
Comment (cannium) [inline compression/const.go]: - give proper attribution to seaweedfs, or we will be in [big trouble](https://zhuanlan.zhihu.com/p/45653912)
- why this file called `const`?
- actually I personally don't think `IsGzippableFileType` needs 2 return values...

File: meta/client/client.go [hunk 4]
```diff
@@ -54,6 +55,16 @@ type Client interface {
...
 	RemoveGarbageCollection(garbage GarbageCollection) error
+	//freezer
+	CreateFreezer(freezer *Freezer) (err error)
+	ListFreezers(maxKeys int) (retFreezers []Freezer, err error)
+	ListFreezersNeedContinue(maxKeys int, status Status) (retFreezers []Freezer, err error)
+	GetFreezer(bucketName, objectName, version string) (freezer *Freezer, err error)
+	GetFreezerStatus(bucketName, objectName, version string) (freezer *Freezer, err error)
+	UploadFreezerStatus(bucketName, objectName string, status, statusSetting Status) (err error)
+	UploadFreezerBackendInfo(targetFreezer *Freezer) error
+	DeleteFreezer(bucketName, objectName string, tx DB) (err error)
```
Comment (cannium) [inline meta/client/client.go]: It seems some of the functions never get called?

File: integrate/yig.sql [hunk 5]
```diff
@@ -200,6 +200,53 @@ CREATE TABLE `objects` (
...
+  `pool` varchar(255) DEFAULT NULL,
+  `ownerid` varchar(255) DEFAULT NULL,
+  `size` bigint(20) DEFAULT NULL,
+  `objectid` varchar(255) DEFAULT NULL,
+  `etag` varchar(255) DEFAULT NULL,
+  `initializationvector` blob DEFAULT NULL,
+  UNIQUE KEY `rowkey` (`bucketname`,`objectname`,`version`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
```
Comment (cannium) [inline integrate/yig.sql]: What if these 2 tables are placed in a new database? Would that lead to a cleaner relationship between `yig` and `yig-restore`? 
@fireworkmarks & @cxt90730

Comment (fireworkmarks) [reply integrate/yig.sql]: If you need to divide into other libraries, you need an additional database connection

File: api/datatype/restore.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	Tier    string   `xml:"Tier"`
+}
+
+func GetRestoreInfo(r *http.Request) (*Restore, error) {
+	restoreInfo := &Restore{}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
```
Comment (cannium) [inline api/datatype/restore.go]: defer r.Body.Close()

File: plugins/snappy_plugin.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	PluginType: COMPRESS_PLUGIN,
+	Create:     GetCompressClient,
+}
+
+func GetCompressClient(config map[string]interface{}) (interface{}, error) {
+	snappy := SnappyCompress{}
+	return snappy, nil
+}
+
+type SnappyCompress struct{}
```
[the remark this answers is not in the document]
Comment (fireworkmarks) [reply plugins/snappy_plugin.go]: I‘ll resolve it

**Pull request merged**